Refine overlay edit button UI: move to link variant and add tooltip

getdave opened the pull request:
## What

Refines the overlay template selector UI by moving the edit button to a more streamlined position and adding a tooltip for improved clarity.

## Why

The previous implementation had the edit button as a large isolated button below the input, which created visual separation and took up unnecessary space. Moving it to a link variant positioned to the right of the control creates a more compact and intuitive UI. Additionally, the button's purpose wasn't immediately clear to all users since it only showed "[Edit]" as visible text, with additional context only available via aria-label for screen readers.

## How

- Moved the edit button from below the SelectControl to a link variant positioned to the right of the control, providing a more streamlined UI
- Added a tooltip to the button that displays the same information as the aria-label (e.g., "Edit overlay: [title]" or "Edit overlay"), making the button's purpose clear to all users
- Extracted the aria-label text into a reusable `editButtonLabel` memoized value for both the tooltip and aria-label
- The aria-label is retained to ensure screen readers continue to receive the full context






## Screenshot
| Before | After |
|--------|-------|
| <img width="598" height="336" alt="Screenshot 2025-12-11 at 16 26 51" src="https://github.com/user-attachments/assets/b9cb5e18-b2f3-4ef7-a31e-3da0a158932b" /> | <img width="274" height="118" alt="Screenshot 2025-12-16 at 13 07 37" src="https://github.com/user-attachments/assets/88abe2e3-146c-4fe7-8a1e-69db888e5b7e" /> |

## Testing

- Enable experiment
- Open Nav block sidebar "Overlay" controls
- See new Edit button position
- Confirm same functionality as per `trunk`.

getdave committed:
Update button style and text

getdave committed:
Add tooltip

youknowriad commented:
This is better for me. I wonder if we have similar patterns in other places that we can align on.

mikachan commented on packages/block-library/src/navigation/editor.scss:
```diff
@@ -650,7 +650,14 @@ body.editor-styles-wrapper .wp-block-navigation__responsive-container.is-menu-op
+}
+
 .wp-block-navigation__overlay-edit-button {
 	margin-top: $grid-unit-10;
```
I don't think `margin-top` is needed if the button is absolutely positioned, or we should instead adjust the top positioning rather than using margin.

mikachan reviewed (commented):
This looks good and tests well for me. 

However, I'm struggling to find any other similar buttons to this design, other than the example @jasmussen gave. 

I'm not opposed to this design, but another option may be to use a hamburger/dropdown menu to contain the Edit link, similar to the one for the Menu tools:

<img width="297" height="274" alt="image" src="https://github.com/user-attachments/assets/551278d1-b660-45d7-8b1e-61900c5460e3" />

I think I prefer the simple "Edit" link though, and I like that it's a clear action rather than being hidden behind another menu. Not strongly opinionated either way, and will defer to design on this!

getdave commented:
<img width="270" height="128" alt="Screenshot 2025-12-15 at 13 48 43" src="https://github.com/user-attachments/assets/3acf832b-d2e5-4083-a639-3ce176d0e732" />

Engineering issue (designers can ignore): Attempting to implement this I got to the above. The problem is that I'm having to artificially add the `Edit` button outside the form control component which means that the help text doesn't span the full width.

I tried the `suffix` prop of `SelectControl` but that replaces the select control dropdown which is not what we want.

I think we should offer greater control over these form components by being able to pass custom `Children`.

youknowriad commented:
@getdave I'd rather accept that the help text doesn't span than hack the component If the UI component is not meant to offer this UI.

getdave committed:
Refine overlay edit button UI: use HStack with flex-start alignment

- Replace absolute positioning with HStack/FlexBlock/FlexItem layout
- Change button variant from link to secondary
- Add 24px top margin to align button with SelectControl input
- Remove absolute positioning CSS rules
- Use CSS variables for spacing

getdave committed:
Shorten overlay selector help text to prevent wrapping

Change help text from 'Select an overlay to use for the navigation.'
to 'Select an overlay for navigation.' to avoid text wrapping
and improve UI appearance.

getdave committed:
Update test to match shortened help text

Update the test expectation to match the new shorter help text
'Select an overlay for navigation.' instead of the previous
longer version.

MaggieCabrera reviewed (approved):
This is working as intended and the code makes sense to me. I'm approving tentatively but I'm unsure of the design. I googled how to translate Edit into German and the word is quite much longer, I'm not sure of this design pattern personally

jeryj commented on packages/block-library/src/navigation/edit/overlay-template-part-selector.js:
```diff
@@ -152,47 +146,42 @@ export default function OverlayTemplatePartSelector( {
+						help={
+							overlayTemplateParts.length === 0 && hasResolved
+								? __( 'No overlays found.' )
+								: __( 'Select an overlay for navigation.' )
```
We should update this text in a follow-up. "Select an overlay for navigation." doesn't really explain to what the selection is going to do. It seems clear that I'm selecting an overlay, but I don't know what it's going to. I'd appreciate more context about that, like, "Customize the overlay appearance"

jeryj commented on packages/block-library/src/navigation/edit/overlay-template-part-selector.js:
```diff
@@ -137,10 +117,24 @@ export default function OverlayTemplatePartSelector( {
+		! hasResolved ||
+		! selectedTemplatePart ||
 		! onNavigateToEntityRecord ||
 		isResolving;
```
Do we need ! hasResolved and isResolving? Also, the Edit button only renders in this condition: `{ overlay && hasResolved && selectedTemplatePart && (` so maybe we can just put this directly in the button?

```
aria-disabled={ onNavigateToEntityRecord }
```

getdave replied:
Yeh some of this disable logic might be redundant now the design has changed.

jeryj commented on packages/block-library/src/navigation/edit/overlay-template-part-selector.js:
```diff
@@ -152,47 +146,42 @@ export default function OverlayTemplatePartSelector( {
+							disabled={ isEditButtonDisabled }
+							accessibleWhenDisabled
+							aria-label={ editButtonLabel }
+							label={ editButtonLabel }
```
`label` already sets the `aria-label`. You can remove the `aria-label`.

jeryj commented on packages/block-library/src/navigation/editor.scss:
```diff
@@ -650,7 +650,10 @@ body.editor-styles-wrapper .wp-block-navigation__responsive-container.is-menu-op
 .wp-block-navigation__overlay-edit-button {
-	margin-top: $grid-unit-10;
-	width: fit-content;
+	margin-top: $grid-unit-30;
```
Could `<Flex/>` or `<HStack/>` handle this layout instead of adding CSS?

getdave replied:
No not with the HTML forced upon us by form components. This was a last resort.

jeryj reviewed (approved):
Approving with suggestions for cleanup. I don't see any blocking issues.

getdave commented:
Let's get these conflicts resolved and cleanup the final points that @jeryj suggested. We should bring this in.

jeryj committed:
Fix duplicate css rule from rebase

jeryj committed:
Remove unnecessary aria-label

jeryj committed:
Refactor edit button disabled code

jeryj committed:
Update unit test
